lib/dynamo/dynamo/jobs.py

```diff
@@ -122,18 +122,36 @@ def _get_credit_cost(job: dict, costs: list[dict]) -> Decimal:
             if cost_definition.keys() not in ({'job_type', 'cost_parameter', 'cost_table'}, {'job_type', 'cost'}):
                 raise ValueError(f'Cost definition for job type {job_type} has invalid keys: {cost_definition.keys()}')
 
-            if 'cost_parameter' in cost_definition:
-                cost_parameter = cost_definition['cost_parameter']
-                parameter_value = job['job_parameters'][cost_parameter]
-                for item in cost_definition['cost_table']:
-                    if item['parameter_value'] == parameter_value:
-                        return item['cost']
-                raise ValueError(f'Cost not found for job type {job_type} with {cost_parameter} == {parameter_value}')
-
-            return cost_definition['cost']
+            return _get_cost_from_definition(job, cost_definition)
+
     raise ValueError(f'Cost not found for job type {job_type}')
 
 
+def _get_cost_from_definition(job, cost_definition):
+    job_type = job['job_type']
+
+    if 'cost' in cost_definition:
+        return cost_definition['cost']
+
+    if 'cost_parameter' in cost_definition:
+        cost_parameter = cost_definition['cost_parameter']
+        parameter_value = job['job_parameters'][cost_parameter]
+
+        if isinstance(parameter_value, list):
+            parameter_value = len(parameter_value)
+
+        for item in cost_definition['cost_table']:
+            if item['parameter_value'] != parameter_value:
+                continue
+
+            if 'cost_table' in item:
+                return _get_cost_from_definition(job, item['cost_table'])
+            else:
+                return item['cost']
```

Comment:
```suggestion
            if 'cost_table' in item:
                return _get_cost_from_definition(job, item['cost_table'])

            if item['parameter_value'] == parameter_value:
                return item['cost']
```

Reply:
This doesn't work because it will always pull the first cost table it sees, regardless of it's parameter value.